test/export/InstanceExporter.test.ts

```diff
@@ -924,6 +924,64 @@ describe('InstanceExporter', () => {
       });
     });
 
+    it('should fix a reference to a type based on a profile', () => {
+      const profiledObservation = new Profile('USCoreObservation');
+      profiledObservation.parent =
+        'http://hl7.org/fhir/us/core/StructureDefinition/us-core-observation-lab';
+      doc.profiles.set(profiledObservation.name, profiledObservation);
+
+      const profiledInstance = new Instance('MyExampleObservation');
+      profiledInstance.instanceOf = 'USCoreObservation';
+      const fixedRefRule = new FixedValueRule('subject');
+      fixedRefRule.fixedValue = new FshReference('Bar');
+      profiledInstance.rules.push(fixedRefRule);
+      doc.instances.set(profiledInstance.name, profiledInstance);
+      doc.instances.set(patientInstance.name, patientInstance);
+      const exported = exportInstance(profiledInstance);
+      expect(exported.subject).toEqual({
+        reference: 'Patient/Bar'
+      });
+    });
+
+    it('should log an error when an invalid reference is fixed', () => {
+      const observationInstance = new Instance('TestObservation');
+      observationInstance.instanceOf = 'Observation';
+      doc.instances.set(observationInstance.name, observationInstance);
+      const fixedRefRule = new FixedValueRule('contact[0].organization');
+      fixedRefRule.fixedValue = new FshReference('TestObservation');
+      // * contact[0].organization = Reference(TestObservation)
+      patientInstance.rules.push(fixedRefRule);
+      doc.instances.set(patientInstance.name, patientInstance);
+
+      const exported = exportInstance(patientInstance);
+      expect(exported.contact).toEqual(undefined); // Contact is not set with invalid type
+      expect(loggerSpy.getAllMessages('error')).toHaveLength(1);
+      expect(loggerSpy.getLastMessage('error')).toMatch(
+        /The type "Reference\(Observation\)" does not match any of the allowed types\D*/s
+      );
+    });
+
+    it('should log an error when fixing an invalid reference to a type based on a profile', () => {
+      loggerSpy.reset();
+      const profiledObservation = new Profile('USCoreObservation');
+      profiledObservation.parent =
+        'http://hl7.org/fhir/us/core/StructureDefinition/us-core-observation-lab';
+      doc.profiles.set(profiledObservation.name, profiledObservation);
+
+      const profiledInstance = new Instance('MyExampleObservation');
+      profiledInstance.instanceOf = 'USCoreObservation';
+      const fixedRefRule = new FixedValueRule('subject');
+      fixedRefRule.fixedValue = new FshReference('Pow'); // Pow is instanceOf Bundle and not an allowed subject reference type
```

Comment:
Isn't `Bundle` not allowed on an `Observation.subject` anyway? Would it make more sense here to test something that is allowed on `Observation.subject` but is then disallowed by the profile? To me it looks like this would be `Group`, `Device`, or `Location`.

Reply:
That's a good idea. I had added this test knowing that `Bundle` was not allowed but thought it made sense for completeness sake, but I think you're right that it can test one of those other three types. I'll update.